### PR TITLE
add test case for #7839

### LINKED ISSUE #7839: Compiler crashes when using generics syntax in type definition
When compiling the following code (tested on 0.18.1 windows amd64 (git 34029263725ef931863ea73966dc08588758dada) and on nim playground) the compilation fails without an error.
I don't know if the syntax is ok though.

```nim
# the following code will break compilation without any error or hint
# `exited with code=3221225725`
type A[I: SomeOrdinal, E] = tuple # same for object
  length: int
 
echo A.sizeof # works without the following proc
 
proc newA*[I: SomeOrdinal, E](): A[I, E] = # works without `SomeOrdinal`
  discard
```

Update: the problem is solved, if `proc newA*[I: SomeOrdinal, E](): A[I, E]` is replaced by `proc newA*[I: SomeOrdinal, E](): A`, so it has nothing to do with `someOrdinal` as thought before.

### PR DESCRIPTION
**Pull request by bung87**
close #7839

### CODE REVIEW

**Comment by Araq** [comment]
Why is the CI red?

File: tests/generics/t7839.nim [hunk 1]
```diff
@@ -0,0 +1,14 @@
+discard """
+action: compile
```
Comment (ringabout) [inline tests/generics/t7839.nim:2]:
```suggestion
action: run
```

File: tests/generics/t7839.nim [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+"""
+# the following code will break compilation without any error or hint
+# `exited with code=3221225725`
+type A[I: SomeOrdinal, E] = tuple # same for object
+  length: int
+ 
+echo A.sizeof # works without the following proc
```
Comment (ringabout) [inline tests/generics/t7839.nim]:
```suggestion
doAssert A.sizeof == 8 # works without the following proc
```

File: tests/generics/t7839.nim [hunk 3]
```diff
@@ -0,0 +1,14 @@
+discard """
+action: compile
+"""
```
Comment (ringabout) [inline tests/generics/t7839.nim]:
```suggestion
```

**Review by ringabout — CHANGES_REQUESTED** [review]

File: tests/generics/t7839.nim [hunk 4]
```diff
@@ -0,0 +1,11 @@
+# the following code will break compilation without any error or hint
+# `exited with code=3221225725`
+type A[I: SomeOrdinal, E] = tuple # same for object
+  length: int
+ 
+doAssert A.sizeof == 8 # works without the following proc
```
Comment (metagn) [inline tests/generics/t7839.nim]:
```suggestion
doAssert A.sizeof == sizeof(int) # works without the following proc
```

**Comment by metagn** [comment]
This seems uncontroversial at this point, just remake as a new PR

**Comment by bung87** [comment]
replaced by https://github.com/nim-lang/Nim/pull/22006

**Pull request closed without merging**